### PR TITLE
Require course seat upgrade deadline

### PR DESCRIPTION
**Pull request by mjfrey**
LEARNER-1021

### CODE REVIEW

**Review by clintonb — APPROVED** [review]

File: ecommerce/static/js/models/course_seats/course_seat.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ define([
 
                     // No validation is needed for empty values or seats not linked to courses.
                     if (_.isEmpty(value) || !course) {
+                        // Do noy allow empty value for verified seats.
```
Comment (rlucioni) [inline ecommerce/static/js/models/course_seats/course_seat.js]: s/noy/not

File: ecommerce/static/js/models/course_seats/course_seat.js [hunk 2]
```diff
@@ -34,6 +34,10 @@ define([
...
+                        if (this.getSeatType() == 'verified') {
+                            return gettext('Upgrade Dealine Required');
```
Comment (rlucioni) [inline ecommerce/static/js/models/course_seats/course_seat.js]: s/Dealine/Deadline

I'd make this a sentence for clarity, too: "Verified seats must have an upgrade deadline."

**Review by rlucioni — APPROVED** [review]
LGTM once test is fixed.

**Comment by rlucioni** [comment]
Please remember to remove the text that appears under the field that refers to a 10-day default.

**Comment by mjfrey** [comment]
@rlucioni Can you take another look?  Especially at the tests.  Want to make sure they are done correctly.

**Review by rlucioni — APPROVED** [review]
:shipit:

**Comment by vkaracic** [comment]
Did anyone test this locally? After this change I cannot create a verified seat anymore no matter what value I put for the `Upgrade deadline` field. Doesn't work on stage either. You can try with a new course that I created `course-v1:edX+VK+01`.

Either I'm doing something wrong or there should be a bug ticket with higher priority filed for this bug since it does not allow creating verified seats anymore.

**Pull request merged**